### ISSUE TITLE
Incompatability with LethalThings and something else?

### ISSUE THREAD

**Issue by ChooMcoo** [Shaosil/LethalCompanyMods-GeneralImprovements#36]
This error is spammed to high hell while I have LethalThings enabled
![image](https://github.com/Shaosil/LethalCompanyMods-GeneralImprovements/assets/155716855/ca86cb78-8103-41e6-b670-2583ac17e875)
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
GeneralImprovements.Patches.PlayerControllerBPatch.SetHoverTipAndCurrentInteractTrigger (GameNetcodeStuff.PlayerControllerB __instance) (at <fe3c77c8ae7b4c04bb669a4ec4a30364>:IL_0112)
(wrapper dynamic-method) GameNetcodeStuff.PlayerControllerB.DMD<GameNetcodeStuff.PlayerControllerB::SetHoverTipAndCurrentInteractTrigger>(GameNetcodeStuff.PlayerControllerB)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Trampoline<GameNetcodeStuff.PlayerControllerB::SetHoverTipAndCurrentInteractTrigger>?274205614(GameNetcodeStuff.PlayerControllerB)
LethalThings.PouchyBelt.PlayerControllerB_SetHoverTipAndCurrentInteractTrigger (On.GameNetcodeStuff.PlayerControllerB+orig_SetHoverTipAndCurrentInteractTrigger orig, GameNetcodeStuff.PlayerControllerB self) (at O:/github/LethalThings/LethalThings/MonoBehaviours/PouchyBelt.cs:81)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Hook<GameNetcodeStuff.PlayerControllerB::SetHoverTipAndCurrentInteractTrigger>?568086116(GameNetcodeStuff.PlayerControllerB)
(wrapper dynamic-method) GameNetcodeStuff.PlayerControllerB.DMD<GameNetcodeStuff.PlayerControllerB::LateUpdate>(GameNetcodeStuff.PlayerControllerB)

When I disable LethalThings I now get this error spammed to high hell
![image](https://github.com/Shaosil/LethalCompanyMods-GeneralImprovements/assets/155716855/a3058919-b3a7-444f-ab90-40fe53e8091e)
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
GeneralImprovements.Patches.PlayerControllerBPatch.SetHoverTipAndCurrentInteractTrigger (GameNetcodeStuff.PlayerControllerB __instance) (at <fe3c77c8ae7b4c04bb669a4ec4a30364>:IL_0112)
(wrapper dynamic-method) GameNetcodeStuff.PlayerControllerB.DMD<GameNetcodeStuff.PlayerControllerB::SetHoverTipAndCurrentInteractTrigger>(GameNetcodeStuff.PlayerControllerB)
(wrapper dynamic-method) GameNetcodeStuff.PlayerControllerB.DMD<GameNetcodeStuff.PlayerControllerB::LateUpdate>(GameNetcodeStuff.PlayerControllerB)

**Comment by Shaosil** [comment]
Hmm, do you have the assets file included in my release in the same location as the DLL? Either way though, I'll look into this and make sure I fix any null references before next release.

**Comment by ChooMcoo** [comment]
Yeah it's in the same folder; installed via r2modman. Also the extra monitors are still blank when I start the game, probably should have mentioned that lol

**Comment by ChooMcoo** [comment]
Oh actually there's no health station either...

**Comment by ChooMcoo** [comment]
I'm managing to narrow down the mod incompatability, standby

**Comment by ChooMcoo** [comment]
Found it! But how peculiar of a mod to be incompatible. I've narrowed it down to https://thunderstore.io/c/lethal-company/p/GuysNight/Balanced_Items/

**Comment by Shaosil** [comment]
Huh that's strange, I'll check it out, thanks for doing the research.

**Comment by Shaosil** [comment]
I used your profile code to reproduce the issue and dug in to this a bit. Unfortunately, this is happening because my StartOfRound.Start() patch is just not being called, leading to several things being uninitialized and null. I'm not sure what it is about Balanced Items being on that causes Harmony to skip my postfix, but I'm not sure what to do about it.

For now, I did add code to successfully suppress that spammy error you were seeing, but until this is fixed or you disable balanced items, certain things about my mod will randomly not function (ship cam pointing north, extra monitors, and med station to name a few)

Sorry, closing as not planned for now.
